### PR TITLE
[DropdownMenu] add helperText & errorText to DropdownMenu Widget

### LINKED ISSUE #119743: Missing `errorText` and `helperText` parameters in M3 `DropdownMenu`
## Steps to Reproduce

1. Add a DropdownMenu widget
2. Try to add errorText and helperText

**Expected results:** 
Show an error message and supporting text, similar to the DropdownButton.

**Actual results:**
The parameters for error and helper style are inside the InputDecorationTheme class, but not the parameters for the texts.

### PR DESCRIPTION
**Pull request by piedcipher**
Adds 2 new properties `helperText` & `errorText` to `DropdownMenu`.

Fixes: #119743

| helperText Dropdown Closed | errorText Dropdown Closed |
| --- | --- |
| photo_2023-03-30_20-21-07 | photo_2023-03-30_20-21-06 |

| helperText Dropdown Open | errorText Dropdown Open |
| --- | --- |
| photo_2023-03-30_20-21-04 | photo_2023-03-30_20-21-02 |

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/dropdown_menu_test.dart [hunk 1]
```diff
@@ -1110,6 +1110,79 @@ void main() {
...
+    await tester.tap(find.byType(DropdownMenu<TestMenu>));
+    await tester.pumpAndSettle();
+    await tester.tap(find.widgetWithText(MenuItemButton, 'Item 3').last);
+    await tester.pumpAndSettle();
+    expect(selectedValue?.label, 'Item 3');
+    expect(node.label, '');
+    expect(node.value, 'Item 3');
+  });
+
+  testWidgets('helperText is not visible when errorText is not null', (WidgetTester tester) async {
```
Comment (QuncCccccc) [inline packages/flutter/test/material/dropdown_menu_test.dart:1157]: Nice test! Can we also add two more tests to check if `DropdownMenu` can respect helperText and errorText when they are not null?

**Review by QuncCccccc — COMMENTED** [review]
Thanks a lot for your contribution! Just left a few comments, but overall it looks great:)

File: packages/flutter/test/material/dropdown_menu_test.dart (hunk 1 above)
Comment (piedcipher) [reply packages/flutter/test/material/dropdown_menu_test.dart:1157]: Done 👍✅

Thanks a lot for the review!

**Review by QuncCccccc — APPROVED** [review]
LGTM. Thanks:)

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 2]
```diff
@@ -183,6 +185,23 @@ class DropdownMenu<T> extends StatefulWidget {
...
+  ///
+  /// Defaults to null;
+  final String? helperText;
+
+  /// Text that appears below the input field and the border to show the error message.
+  ///
+  /// If non-null, the border's color animates to red and the [helperText] is not shown.
+  ///
+  /// Defaults to null;
+  final String? errorText;
```
Comment (justinmc) [inline packages/flutter/lib/src/material/dropdown_menu.dart:203]: Nit: For both of these new parameters, would it be useful to link to the corresponding InputDecorator property or is that too much detail?

```
/// See also:
///
///  * [InputDecorator.errorText]
```

**Review by justinmc — APPROVED** [review]
LGTM :+1: 

Thanks for the thorough tests here.

File: packages/flutter/lib/src/material/dropdown_menu.dart (hunk 2 above)
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/dropdown_menu.dart:203]: Agree with @justinmc :) Might be more clear to users if we can put this info there. I think we can do something like this:
```dart
///
/// See also:
/// 
/// * [InputDecoration.errorText], which is the text that appears below the [InputDecorator.child] and the border.
```

For helper text, we can put something like:
```dart
///
/// See also:
/// 
/// * [InputDecoration.helperText], which is the text that provides context about the [InputDecorator.child]'s value.
```

**Comment by piedcipher** [comment]
> The Google testing is failing. I don't think the failure is related to the code change. We might need to rebase master and re-submit to trigger the test again:) Thanks!

Sure I'll rebase, but it failed for `DanTup` as well.
Screenshot_20230405-231828

**Comment by piedcipher** [comment]
> The Google testing is failing. I don't think the failure is related to the code change. We might need to rebase master and re-submit to trigger the test again:) Thanks!

I did rebase, I don't know how these other commits got added :(

I used GitHub's UI (1 click) for rebase.

**Comment by piedcipher** [comment]
> auto label is removed for flutter/flutter, pr: 123775, due to - The status or check suite [Google testing](http://flutter-rob/#/flutter/presubmit?pr=123775) has failed. Please fix the issues identified (or deflake) before re-applying this label.

This check is failing on all PRs at the moment.

**Comment by piedcipher** [comment]
Can we please add auto-submit lable?

**Comment by piedcipher** [comment]
Number of checks are 8 now, earlier there were 71.

**Comment by piedcipher** [comment]
> auto label is removed for flutter/flutter, pr: 123775, Failed to merge pr#: 123775 with Pull request could not be merged: Required status check "ci.yaml validation" is expected..

Can I please get auto-submit label again? Sorry to bother again. Previous failure was due to an infra issue #124332.

**Comment by justinmc** [comment]
No worries.  It looks like the Google tests are running now though.

**Pull request merged**